backends/ebpf/ebpf_runtime.c

```diff
@@ -1,60 +1,77 @@
 #include <string.h>
+#include <libgen.h>
 #define PCAP_DONT_INCLUDE_PCAP_BPF_H
 #include <pcap/pcap.h>
 #include "test.h"
 
+
+void usage() {
+    printf("This program parses a pcap file, feeds the individual packets into a filter function, and returns the output.\n");
+}
+
 int main(int argc, char **argv) {
     pcap_t *in_handle;
     pcap_dumper_t *out_handle;
     const unsigned char *packet;
     char errbuf[PCAP_ERRBUF_SIZE];
     struct pcap_pkthdr *pcap_hdr;
-
+    int ret;
     /* Skip over the program name. */
     ++argv; --argc;
     if (argc != 1) {
-        fprintf(stderr, "Missing input trace file, exiting...\n");
+        usage();
+        fprintf(stderr, "The input trace file is missing."
+                 "Expected .pcap file as input argument.\n");
         return EXIT_FAILURE;
     }
 
     /* Initialize the registry of shared tables */
     struct bpf_map_def* current = tables;
     while (current->name != 0) {
-         if (BPF_OBJ_PIN(current, current->name) != 0) {
-            fprintf(stderr, "Error: Failed to register table.\n");
-            return EXIT_FAILURE;
-         }
+        registry_add(current);
         current++;
     }
 
     /* Open and read pcap file */
     in_handle = pcap_open_offline(argv[0], errbuf);
     if (in_handle == NULL) {
-        fprintf(stderr, "Error: Failed to read pcap file: %s\n", errbuf);
+        pcap_perror(in_handle, "Error: Failed to read pcap file ");
         return EXIT_FAILURE;
     }
 
     /* Create the output file. */
-    out_handle = pcap_dump_open(in_handle, "out.pcap");
+    char *in_file = strdup(argv[0]);
+    char *in_dir = dirname(in_file);
+    int in_dir_len = strlen(in_dir);
+    /* length of the input directory plus the expected length of the pcap name */
+    char *out_file = calloc(in_dir_len + strlen("/pcapxxxx_out.pcap") , 1);
```

Comment:
You should probably have a constant string and use it everywhere instead of repeating it.
Also, your strings are not the same.